politeiawww/backend.go

```diff
@@ -2873,72 +2865,28 @@ func (b *backend) ProcessVoteStatus(token string) (*www.VoteStatusReply, error)
 	}, nil
 }
 
-// ProcessUserCommentsVotes returns the votes an user has for the comments of a given proposal
-func (b *backend) ProcessUserCommentsVotes(user *database.User, token string) (*www.UserCommentsVotesReply, error) {
-	log.Tracef("ProcessUserCommentsVotes")
-
-	payload, err := decredplugin.EncodeGetProposalCommentsVotes(decredplugin.GetProposalCommentsVotes{
-		Token: token,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	// Obtain proposal comments votes from plugin
-	challenge, err := util.Random(pd.ChallengeSize)
-	if err != nil {
-		return nil, err
-	}
-
-	pc := pd.PluginCommand{
-		Challenge: hex.EncodeToString(challenge),
-		ID:        decredplugin.ID,
-		Command:   decredplugin.CmdProposalCommentsVotes,
-		CommandID: decredplugin.CmdProposalCommentsVotes,
-		Payload:   string(payload),
-	}
+// ProcessUserCommentsLikes returns the votes an user has for the comments of a given proposal
+func (b *backend) ProcessUserCommentsLikes(user *database.User, token string) (*www.UserCommentsLikesReply, error) {
+	log.Tracef("ProcessUserCommentsLikes")
 
-	responseBody, err := b.makeRequest(http.MethodPost,
-		pd.PluginCommandRoute, pc)
-	if err != nil {
-		return nil, err
-	}
-
-	var reply pd.PluginCommandReply
-	err = json.Unmarshal(responseBody, &reply)
-	if err != nil {
-		return nil, fmt.Errorf("Could not unmarshal "+
-			"PluginCommandReply: %v", err)
-	}
-
-	// Verify the challenge.
-	err = util.VerifyChallenge(b.cfg.Identity, challenge, reply.Response)
-	if err != nil {
-		return nil, err
-	}
-
-	gpcvr, err := decredplugin.DecodeGetProposalCommentsVotesReply([]byte(reply.Payload))
-	if err != nil {
-		return nil, err
-	}
+	userID := user.ID.String()
+	uclr := www.UserCommentsLikesReply{}
+	b.RLock()
+	defer b.RUnlock()
+	for k, v := range b.userLikeActionByCommentID {
```

Comment:
This probably needs to go into a map. Do we have an idea how many iterations we do here?

Reply:
It is a map, this is how it is defined:
```Go
// User vote action on each comment
userLikeActionByCommentID map[string]map[string]int64 // [token+userid][commentid]action
```
The number of iterations is equals to the number of comment votes a user has given to a specific proposal.

Reply:
I am talking about the for loop. What is a normal amount of iterations?

Reply:
It depends on fo the number of users and proposals. The worse scenario is `NU x NP`,  being NU the total number of users and NP the total number of proposals.

Reply:
From slack resolution:
```
fernandoabolafio [3:18 PM]
ahh, okay, that makes sense, thanks
2nd one:
You asked me how many iterations we could be doing for:
`for k, v := range b.userLikeActionByCommentID {...}`
So, the worst scenario here is a number of iterations equals to `totalNumberOfUsers x totalNumberOfProposals`. For the current live Pi I would estimate something about  1300  interactions (in the worst case) considering 13 proposals and 100 users.
Is that your question?

moo31337 [3:31 PM]
is this a one time deal?

fernandoabolafio [3:33 PM]
no, it happens as a result of an user request for their comment likes for a given proposal. E.g every time a logged in user access a proposal route.

moo31337 [3:34 PM]
so can we not iterate 1300 times?
that is an expensive lock

fernandoabolafio [3:35 PM]
Yeah, I can optimize that
```